src/main/java/org/fundacionjala/coding/kevinherrera/crypto/CesarCrypto.java

```diff
@@ -0,0 +1,133 @@
+package org.fundacionjala.coding.kevinherrera.crypto;
+
+/**
+ * Class to encode a message using a key.
+ */
+public class CesarCrypto {
+
+    private static final  int CHAR_Z_UPPER_HASHCODE = 90;
+    private static final  int CHAR_A_UPPER_HASHCODE = 65;
+
+    /**
+     * Enum encrypt tasks.
+     */
+    enum EncryptTask { ENCODE, DECODE }
+
+    /**
+     * codifies the alphabetic values of a message using a key.
+     * @param message message to encode
+     * @param key key to encode
+     * @return message as UPPERCASE type encoded
+     */
+    public String encode(final String message, final int key) {
+        return processEncrypt(EncryptTask.ENCODE, message, key);
+    }
+
+    /**
+     * decode the alphabetic values of a message that was encode using a key.
+     * @param encodeMessage message to decode
+     * @param key key to decode
+     * @return message as lowercase type decoded
+     */
+    public String decode(final int key, final String encodeMessage) {
+        return processEncrypt(EncryptTask.DECODE, encodeMessage, key).toLowerCase();
+    }
+
+    /**
+     * encode the alphabetic values of a message using vigenere method.
+     * @param message message to encode
+     * @param key key to encode
+     * @return message as UPPERCASE type encode
+     */
+    public String encodeVigenere(final String message, final String key) {
+        return processEncryptVigenere(EncryptTask.ENCODE, message, key);
+    }
+
+    /**
+     * decode the alphabetic values of a message encoded with vigenere key.
+     * @param message message to decode
+     * @param key key to decode
+     * @return message as lowercase type decoded
+     */
+    public String decodeVigenere(final String key, final String message) {
+        return processEncryptVigenere(EncryptTask.DECODE, message, key).toLowerCase();
+    }
+
+    /**
+     * internal process of encrypt functions.
+     * @param task encrypt task
+     * @param message message to process
+     * @param key codification key
+     * @return message encrypted or decrypted determinated by task value
+     */
+    private String processEncrypt(final EncryptTask task, final String message, int key) {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)